src/main/java/org/broadinstitute/hellbender/utils/fasta/CachingIndexedFastaSequenceFile.java

```diff
@@ -94,26 +77,21 @@ public CachingIndexedFastaSequenceFile(final Path fasta, final FastaSequenceInde
      * @param cacheSize the size of the cache to use in this CachingIndexedFastaReader, must be >= 0
      * @param preserveCase If true, we will keep the case of the underlying bases in the FASTA, otherwise everything is converted to upper case
      */
-    public CachingIndexedFastaSequenceFile(final Path fasta, final long cacheSize, final boolean preserveCase, final boolean  preserveIUPAC) throws FileNotFoundException {
-        super(fasta);
+    CachingIndexedFastaSequenceFile(final Path fasta, final long cacheSize, final boolean preserveCase, final boolean  preserveIUPAC) {
+        final ReferenceSequenceFile referenceSequenceFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(fasta, true, true);
+        sequenceFile = requireIndex(fasta, referenceSequenceFile);
         if ( cacheSize < 0 ) throw new IllegalArgumentException("cacheSize must be > 0");
         this.cacheSize = cacheSize;
         this.cacheMissBackup = Math.max(cacheSize / 1000, 1);
         this.preserveCase = preserveCase;
         this.preserveIUPAC = preserveIUPAC;
     }
 
-    /**
-     * Same as general constructor but allows one to override the default cacheSize
-     *
-     * By default, this CachingIndexedFastaReader converts all incoming bases to upper case
-     *
-     * @param fasta the file we will read our FASTA sequence from.
-     * @param index the index of the fasta file, used for efficient random access
-     * @param cacheSize the size in bp of the cache we will use for this reader
-     */
-    public CachingIndexedFastaSequenceFile(final Path fasta, final FastaSequenceIndex index, final long cacheSize) {
-        this(fasta, index, cacheSize, false, false);
+    private static ReferenceSequenceFile requireIndex(Path fasta, ReferenceSequenceFile referenceSequenceFile) {
+        if( !referenceSequenceFile.isIndexed()) {
+            throw new UserException("Couldn't create indexed fasta: " + fasta.toAbsolutePath());
```

Comment:
The error reporting here is kind of bad, there's no easy way to tell what the problem was.  I added a gross block of error diagnosing code, let me know what you think...

Reply:
Actually, I punted here and moved the error checking all to be in the initial checks.  This throws gatk error now if it ends up here without being indexed since we should be preventing that up front.